### PR TITLE
shellcheck verification and cert-manager installation script

### PR DESCRIPTION
**Pull request by sayantani11**
**What type of PR is this?**

**What this PR does / why we need it**:
This PR adds a bash script to run shellcheck on all the scripts in the repository and fix shellcheck warnings in existing bash scripts.
And make cert-manager install in makefile more resilient
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Shellcheck verification & cert-manager installation script
```

### CODE REVIEW

File: hack/install-cert-manager.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
```
Comment (cpanato) [inline hack/install-cert-manager.sh:1]: where are we using this script?

File: scripts/ci-test-coverage.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
Comment (cpanato) [inline scripts/ci-test-coverage.sh:1]: where we are using this? did you create a PR in test-infra to add run this?

File: scripts/ci-test-coverage.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+# shellcheck source=hack/ensure-go.sh
+source "${REPO_ROOT}/hack/ensure-go.sh"
+
+cd "${REPO_ROOT}" && \
+      make test-cover
```
Comment (cpanato) [inline scripts/ci-test-coverage.sh:24]: we dont have this rule in the makefile.

I would say we remove this file for now and if we need that we add in a separated PR

**Review by cpanato — CHANGES_REQUESTED** [review]
thanks fro this PR

a few questions

File: hack/install-cert-manager.sh (hunk 1 above)
Comment (sayantani11) [reply hack/install-cert-manager.sh:1]: To make cert manager install in makefile more resilient. Didn't notice I forgot to change that in the makefile. Did that now.

**Comment by sayantani11** [comment]
cc @cpanato

**Pull request closed without merging**